### ISSUE TITLE
Daikin AC - Select "Minimum" fan speed, then automatically changes to "Medium-Low"

### ISSUE THREAD

**Issue by MrFusion4** [scottyphillips/echonetlite_homeassistant#185]
Hello,

Firstly, HUGE THANKS to everyone working on this project.  I can't express how useful this integration's been for my smart home setup here in Japan, having all five of my air conditioners (two Mitsubishi, three Daikin) in HA and also fed through to Apple Home app via the HomeKit integration.  I've also just begun using light switches too via the supported Panasonic switch hub, which works perfectly too!

Back to the issue... My Daikin AC units (Risora) support seven fan speeds (Auto, Minimum, Medium-Low, Medium, Medium-High, High, Very-High) which I've verified through use of Daikin's iOS app and the unit's remote control.  When selecting "Minimum" in Daikin's app, the fan speed is selected correctly and remains there.  However, when using this integration, it correctly changes to fan speed "Minimum", but after a period of a few seconds, automatically changes to speed "Medium-Low", (regardless of the previous fan setting.)

I've attached a couple photos of the HA setup and logs to show how it's working.  The bottom two commands were triggered by the iOS app (showing how "Minimum" can be selected and remain selected.)  The commands above are all from HA.

If there's more information I can provide, please let me know how.  I'm a bit tech inclined, but if given instructions, I can follow them.

Thanks again for your continued support!

![1](https://github.com/scottyphillips/echonetlite_homeassistant/assets/119113168/207282ed-eda3-4f8a-9a73-e7180ff01651)
![2](https://github.com/scottyphillips/echonetlite_homeassistant/assets/119113168/fc820580-6e79-4e45-90c6-69d9b191ad92)

**Comment by nao-pon** [comment]
@MrFusion4 The minimum airflow value of the air conditioner may be low. Try adding "low" to your settings and observing.

**Comment by MrFusion4** [comment]
Just tried adding "low" and switching to it.  Unfortunately, it produced the same result.

As I noted before, "minimum" is shown as set correctly in Home Assistant when triggered by the Daikin iOS app.

![11](https://github.com/scottyphillips/echonetlite_homeassistant/assets/119113168/04b0c73a-0695-4956-a4f7-dd61c5e85aba)

**Comment by nao-pon** [comment]
This is a very strange behavior and I don't know the cause. First, please obtain a debug log and let us know.

If you enable the debug log on this integration screen in HA, perform a series of operations, and then disable the debug log, you will be able to download the log. Please upload that log file here.

**Comment by MrFusion4** [comment]
Uploaded as requested.  The first image shows the HA logbook, how I switched the fan speed to Medium, then to Minimum, and then it near instantly changed to Medium-Low.  The second & third images are from HA's log.

![12](https://github.com/scottyphillips/echonetlite_homeassistant/assets/119113168/3c74427f-0c1d-4f90-b2a4-f33277ebce7e)
![13](https://github.com/scottyphillips/echonetlite_homeassistant/assets/119113168/704a42d3-ef23-476d-93e7-d0719b2fa481)
![14](https://github.com/scottyphillips/echonetlite_homeassistant/assets/119113168/659eaaec-565f-4597-899a-def010438e8e)

**Comment by nao-pon** [comment]
@MrFusion4 Can you upload a text file of the log?

**Comment by MrFusion4** [comment]
Sure!

[Log.txt](https://github.com/user-attachments/files/16094180/Log.txt)

**Comment by MrFusion4** [comment]
Interesting!  Thanks for checking.

I ran debug mode again and changed the mode to Minimum using the Daikin iOS app.  I put the logs in the file below, the top portion's the Home Assistant command for Minimum, while the bottom portion's generated from Daikin's app, which successfully makes the change.  Looking at these side-by-side, are there any noticeable differences?

[HA Daikin Log.txt](https://github.com/user-attachments/files/16119009/HA.Daikin.Log.txt)

**Comment by nao-pon** [comment]
If the device's fan speed is set correctly when you change from HA to minimum, then I think it's a bug in the device, because the response to the device's query is incorrect.

On the other hand, if the device's fan speed is not set correctly when you change from HA to minimum, then I think it doesn't support the minimum setting in the econet-lite protocol.

**Comment by nao-pon** [comment]
Daikin's object-specific property declaration for this product can be downloaded from the following page. The declaration has properties divided into two sections, upper and lower, and the range of air volume is different between the upper and lower sections. I don't understand the meaning of this division into upper and lower sections, but this may be related.

[FireShot Capture 018 -  - .pdf](https://github.com/user-attachments/files/17058331/FireShot.Capture.018.-.-.pdf)

- https://echonet.jp/introduce/gz-000855-2/
- https://echonet.jp/wp/wp-content/uploads/obj_prop/GZ/GZ-000855/Implemented_Properties_(0x013001).pdf

**Comment by nao-pon** [comment]
This is most likely a specification limitation, so I'll close this for now, you can reopen it if you find a solution to this problem.

**Comment by MrFusion4** [comment]
Thank you very much for finding that information!  I was able to test your conclusion, and it turns out you're correct.  To do so, I used a different iOS app (Mitsubishi HEMS HM-ST03, which was also the default app for use with my two Mitsubishi units) to control the Daikin units.  Mitsubishi's app allowed for selection of all 8 Echonet protocol specified speeds (listed in the app as "Auto" and "1" through "8".  When I selected "1" ("Minimum" as per spec) the unit immediately defaulted to "3" ("Medium-Low") setting.  So it appears that my unit does not support "Minimum" speed via Echonet protocol, and instead is able to set it only via Diakin's remote control and iOS apps.

Now that I know this information, I've written Daikin customer support.  They still push firmware updates periodically for my units, so I've requested they update their Echonet spec to include the "Minimum" setting next time.

Much appreciated.
